PR-Review-Guide.md

```diff
@@ -0,0 +1,129 @@
+* [High level considerations](#high-level-considerations)
+    * [Security](#security)
+    * [Reliability](#reliability)
+    * [Performance](#performance)
+    * [Maintenance](#maintenance)
+    * [Usability](#usability)
+* [Specifics to consider](#specifics-to-consider)
+    * [Basics](#basics)
+    * [Layout and formatting](#layout-and-formatting)
+    * [Design](#design)
+    * [Code comments](#code-comments)
+    * [Documentation](#documentation)
+    * [Logging and Debugging](#logging-and-debugging)
+    * [Testing](#testing)
+    * [Environments](#environments)
+    * [Upgrades](#upgrades)
+* [Mandatory Checks](#mandatory-checks)
+
+---
+
+This document attempts to capture some of the points to consider when reviewing a [new code contribution](CONTRIBUTING.md) (a PR) to ensure overall quality is kept as high as possible.
+
+## High level considerations
+
+### Security
+
+- Could the change introduce a potential security risk?
+  - What additional checks would resolve it?
+
+### Reliability
+
+- What happens on failure? (Consider **every** possible failure)
```

Comment:
Not sure if you are looking for elaboration notes as well, but I'll drop them in :-)
- might it accidentally leave items around - temp files etc.
- might it leak resources - memory, open files etc.